src/module/actor/sheet/starship.js

```diff
@@ -578,22 +578,20 @@ export class ActorSheetSFRPGStarship extends ActorSheetSFRPG {
         event.preventDefault();
 
         const data = TextEditor.getDragEventData(event);
-        if (!data) return false;
-
-        // Case - Dropped Actor
-        if (data.type === "Actor") {
+        if (Hooks.call('dropActorSheetData', this.actor, this, data) === false) {
+            // Further processing halted
+        } else if (data.type === "Actor") {
             const actor = await Actor.fromDropData(data);
-            return this._onCrewDrop(event, actor.id);
+            await this._onCrewDrop(event, actor.id);
```

Comment:
I'd rather keep these as `return`s for readability's sake. I find it indicates more clearly that there's no more code to execute after the `if-else`s

Reply:
Sure. I don't believe it makes a functional difference, so I'll leave it as it was.

Although, perhaps you can help me out. I made this change because I discovered that [`DocumentSheet._onDrop`](https://foundryvtt.com/api/classes/client.DocumentSheet.html#_onDrop) returns `void`. I'm only realizing now that [`ActorSheet._onDrop`](https://foundryvtt.com/api/classes/client.ActorSheet.html#_onDrop) returns a `Promise<boolean|object>`, but I'm not seeing an explanation of what that return value represents. Do you know what Foundry does (if anything) with this return value?

Reply:
It doesn't make a functional difference, yes, it's just an aesthetics thing.

`_onDrop` or its derivatives like `onDropActor` can return `false` to cancel the drop. The object is the drop data if it's successful. I'd recommend looking at `foundry.js` and `foundry-esm.js` directly instead of the API docs, they're not very informative